src/algorithms/signal_source/gnuradio_blocks/fifo_reader.cc

```diff
@@ -0,0 +1,111 @@
+/*!
+ * \file fifo_reader.cc
+ *
+ * \brief Implementation of the class to retrieve samples from an existing Unix FIFO
+ * \author Malte Lenhart, 2021. malte.lenhart(at)mailbox.org
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * GNSS-SDR is a Global Navigation Satellite System software-defined receiver.
+ * This file is part of GNSS-SDR.
+ *
+ * Copyright (C) 2010-2020  (see AUTHORS file for a list of contributors)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * -----------------------------------------------------------------------------
+ */
+
+#include "fifo_reader.h"
+#include <glog/logging.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
```

Comment:
review your includes and remove the ones not needed. For example, `<fstream>` is probably not needed here, it must be in the class header. `<iostream>` is only needed for the standard streams; prefer glog to cout. That said, there is plenty of cout littered through the code base, so follow the pattern